### PR TITLE
task(ci): add sonar

### PR DESCRIPTION
**Pull request by afflom**
https://sonarcloud.io/summary/overall?id=rh-psce_trestle-bot

### CODE REVIEW

File: .github/workflows/codecov.yml [hunk 1]
```diff
@@ -2,9 +2,12 @@
 name: Code Coverage Check
 
 on:
+  push:
+    branches:
+      - main
   pull_request:
-    branches: 
-    - main
+    types: [opened, synchronize, reopened]
```
Comment (jpower432) [inline .github/workflows/codecov.yml:9]: I believe this line is redundant. These activity types are what are set by default. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request.

File: .github/workflows/codecov.yml [hunk 2]
```diff
@@ -26,4 +29,35 @@ jobs:
...
+        with:
+          args: >
+            -Dsonar.python.coverage.reportPaths=coverage.xml
+            -Dsonar.tests=tests/
+            -Dsonar.sources=trestlebot/ 
+            -Dsonar.python.version=3.10
+            -Dsonar.projectKey=rh-psce_trestle-bot
+            -Dsonar.organization=rh-psce
+      - name: SonarQube Quality Gate check
+        uses: sonarsource/sonarqube-quality-gate-action@master
```
Comment (jpower432) [inline .github/workflows/codecov.yml:59]: Can we pin to a commit instead of the master branch?

File: .github/workflows/codecov.yml [hunk 3]
```diff
@@ -26,4 +29,35 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Get coverage
+        uses: actions/download-artifact@v2
+        with:
+          name: coverage
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@master
```
Comment (jpower432) [inline .github/workflows/codecov.yml:46]: Can we pin to a commit instead of the master branch?

**Review by jpower432 — COMMENTED** [review]
Looks good. A couple questions/comments.

**Review by jpower432 — APPROVED** [review]
LGTM. Do you have test branch showing this running successfully? The workflow ended up being skipped in this instance because there are no python changes.

**Comment by afflom** [comment]
> LGTM. Do you have test branch showing this running successfully? The workflow ended up being skipped in this instance because there are no python changes.

I found a phrase with a period that I removed to verify functionality. PTAL.

**Pull request merged**